internal/api/create_credentials_v1alpha1.go

```diff
@@ -34,34 +34,63 @@ func (s *server) CreateCredentials(
 		return nil, connect.NewError(connect.CodeUnimplemented, errSecretManagementDisabled)
 	}
 
-	creds := credentials{
-		project:        req.Msg.GetProject(),
-		name:           req.Msg.GetName(),
-		description:    req.Msg.GetDescription(),
-		credType:       req.Msg.GetType(),
-		repoURL:        req.Msg.GetRepoUrl(),
-		repoURLIsRegex: req.Msg.GetRepoUrlIsRegex(),
-		username:       req.Msg.GetUsername(),
-		password:       req.Msg.GetPassword(),
+	credType := req.Msg.GetType()
+
+	if err := validateFieldNotEmpty("type", credType); err != nil {
+		return nil, err
 	}
 
+	switch credType {
+	case kargoapi.CredentialTypeLabelValueGit,
+		kargoapi.CredentialTypeLabelValueHelm,
+		kargoapi.CredentialTypeLabelValueImage:
```

Comment:
This is more conventional in Go -- return early if there's a validation error and then the rest of the code isn't in the switch.

```suggestion
	switch credType {
	case kargoapi.CredentialTypeLabelValueGit,
		kargoapi.CredentialTypeLabelValueHelm,
		kargoapi.CredentialTypeLabelValueImage:
	default:
		return nil, connect.NewError(
			connect.CodeInvalidArgument,
			errors.New("type should be one of git, helm, image"),
		)
	}
```

Reply:
Actually... the `validateCredentials()` method is already doing this same check.